src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs

```diff
@@ -526,23 +521,94 @@ internal abstract partial class SafeDeleteContext : SafeHandle
                                             refContext!,
                                             ref outSecurityBufferDescriptor,
                                             ref outFlags,
-                                            outFreeContextBuffer);
+                                            null);
+
+                            if (isSspiAllocated)
+                            {
+                                outoutBuffer = outUnmanagedBuffer.pvBuffer;
+                            }
+
+                            // Get unmanaged buffer with index 0 as the only one passed into PInvoke.
+                            outSecBuffer.size = outUnmanagedBuffer.cbBuffer;
+                            outSecBuffer.type = outUnmanagedBuffer.BufferType;
+                            outSecBuffer.token = outSecBuffer.size > 0 ?
+                                new Span<byte>((byte*)outUnmanagedBuffer.pvBuffer, outUnmanagedBuffer.cbBuffer).ToArray() :
+                                null;
+
+                            if (inSecBuffers.Count > 0 && inUnmanagedBuffer[1].BufferType == SecurityBufferType.SECBUFFER_EXTRA && inSecBuffers._item1.Type == SecurityBufferType.SECBUFFER_EMPTY)
```

Comment:
Is > 0 correct here or should it be > 1?

Reply:
yes, it should be 1. This was added to make sure we use valid buffers. Since we always have three, checking provided count is only one way how to know what has real content. 

As far as the relation: The inSecBuffers as managed struct with type and span e.g length and data. That will get copied info SspiCli.SecBuffer we stackalloc and passed to Schannel. The caveat is that despite the name, this is not IN and Schannel can modify the inUnmanagedBuffer and return back some information. I added the check to be sure that originally passed empty buffer turned into SECBUFFER_EXTRA e.g. unprocessed data. Maybe doing only the second part would be sufficient. AFAK I did not see us passing SECBUFFER_EXTRA in so in theory that can be only result of schannel processing - but I wanted to be sure. I hope it make some more sense.